ui/articles-aem-renderer/src/main/java/org/cru/godtools/articles/aem/fragment/AEMArticleItemFragment.java

```diff
@@ -138,21 +150,72 @@ private String injectCss(@NonNull final String content) {
 
     // endregion WebView content
 
+    //region WebClient
+
     private class AEMWebViewClient extends WebViewClient {
         @Nullable
         @Override
         public WebResourceResponse shouldInterceptRequest(WebView view, String url) {
-            if (url.endsWith(".jpg")) {
-                //TODO: Load local Image WebResource
-            } else if (url.endsWith(".css")) {
-                //TODO: Load CSS WebResource
+            String extension = MimeTypeMap.getFileExtensionFromUrl(url);
+
+            WebResourceResponse response = null;
+
+            String mimeType;
+
+            switch (extension) {
+                case "jpg":
+                case "png":
+                case "bmp":
+                case "gif":
+                    mimeType = String.format("image/%s", "jpg".equals(extension) ? "jpeg" : extension);
+                    response = getResponseFromFile(mimeType, url);
+                    break;
+                case "css":
+                    mimeType = "text/css";
+                    response = getResponseFromFile(mimeType, url);
+                    break;
+            }
+
+            if (response == null) {
+                return super.shouldInterceptRequest(view, url);
```

Comment:
you should return a not found `WebResourceResponse` and not let the WebView trigger an actual web request

Reply:
So you don't want anything to load if we haven't  saved the file also this runs for files that are not covered in the switch statement like ".ico"

Reply:
yeah, I would also change the switch to always attempt to load the resource from disk, use "application/octet-stream" as the mime-type if you don't have a preset mime-type